videostore/renamer.go

```diff
@@ -0,0 +1,131 @@
+type renamer struct {
+	watchDir  string
+	outputDir string
+	logger    logging.Logger
+}
+
+func newRenamer(watchDir, outputDir string, logger logging.Logger) *renamer {
+	return &renamer{
+		watchDir:  watchDir,
+		outputDir: outputDir,
+		logger:    logger,
+	}
+}
+
+// processSegments periodically scans the directory for new files to process
+//
+// This function polls the directory regularly, looking for MP4 files
+// and processing them to convert their filenames from local datetime
+// to unix timestamps. It runs until the context is done, at which point
+// it flushes any remaining files and exits gracefully.
+func (r *renamer) processSegments(ctx context.Context) error {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	r.logger.Debugf("Starting to scan directory: %s", r.watchDir)
+	for {
+		select {
+		case <-ctx.Done():
+			r.logger.Debug("Context done, stopping scanner and flushing remaining files")
+			if err := r.scanAndProcessFiles(); err != nil {
+				r.logger.Errorf("Error flushing directory: %v", err)
+			}
+			return nil
+		case <-ticker.C:
+			if err := r.scanAndProcessFiles(); err != nil {
+				r.logger.Errorf("Error scanning directory: %v", err)
+			}
+		}
+	}
+}
+
+// scanAndProcessFiles scans the watch directory for MP4 files and processes them
+func (r *renamer) scanAndProcessFiles() error {
+	files, err := r.getMPEGFiles()
+	if err != nil {
+		return err
+	}
+
+	return r.processFiles(files)
+
+}
+
+// getMPEGFiles retrieves valid and complete MP4 files in the watch directory
+func (r *renamer) getMPEGFiles() ([]string, error) {
+	entries, err := os.ReadDir(r.watchDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read directory: %w", err)
+	}
+	var mpegFiles []string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			r.logger.Debugf("Skipping directory entry: %s", entry.Name())
+			continue
+		}
+		name := entry.Name()
+		if !strings.HasSuffix(name, ".mp4") {
+			r.logger.Debugf("Skipping non-MP4 file: %s", name)
+			continue
+		}
+		fullPath := filepath.Join(r.watchDir, name)
+		// Check if the video segment is completed with MOOV atom present
+		_, err := getVideoInfo(fullPath)
+		if err == nil {
+			mpegFiles = append(mpegFiles, fullPath)
+		} else {
+			r.logger.Debugf("Not a valid video file, skipping %s: %v", fullPath, err)
+		}
+	}
+
+	return mpegFiles, nil
+}
+
+// processFiles renames a list of files
+func (r *renamer) processFiles(files []string) error {
+	if len(files) == 0 {
+		r.logger.Debug("No files to process")
+		return nil
+	}
+	var lastErr error
+	for _, file := range files {
+		r.logger.Debugf("Processing file: %s", file)
+		if err := r.convertFilenameToUnixTimestamp(file); err != nil {
+			r.logger.Errorf("Failed to process %s: %v", file, err)
+			lastErr = err
+		}
+	}
+
+	return lastErr
+}
+
+// convertFilenameToUnixTimestamp changes filename from local datetime to unix timestamp
+func (r *renamer) convertFilenameToUnixTimestamp(filePath string) error {
```

Comment:
1. This shouldn't spam errors if there are mp4 files which don't follow the expected string format
2. Let's just make this do the formatting if filename is in the expected format and otherwise ignore that file.

Reply:
Good call, if filename schema is incorrect we just debug log and skip now.